### PR TITLE
feat: memory improvements

### PR DESCRIPTION
**Pull request by BrianAmadori**
## What does this PR change?

...

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch={branch_name}&{desired_url_params}
2. ...

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by AjimenezDCL — APPROVED** [review]

**Comment by AjimenezDCL** [comment]
I'm a bit concerned about the amount of complexity we are adding to the project. The global rendering tracking through the `DataStore` or the micro-optimizations can be hard to follow or debug in the future.

Usually performance comes with sacrificies on readability and complexity so it will pay off, but we should keep an eye on documentation to ease knowledge transfer and onboardings. 

Great job!

**Comment by Suduck** [comment]
I am unable to click on the entities in builder in world seems like it is not detecting the collider of the meshes since it is not outlining the items. Let discuss about how we should fix this

**Review by Suduck — CHANGES_REQUESTED** [review]
BIW selecting is not working

**Comment by Suduck** [comment]
I am not sure if you asked for a review because you fixed the biw part, just in case, it is not fixes yet, let me know if you need help with that part

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarRenderer.cs [hunk 1]
```diff
@@ -456,6 +456,7 @@ private IEnumerator LoadAvatar()
             if (mergeSuccess)
             {
                 gpuSkinning = new SimpleGPUSkinning(avatarMeshCombiner.renderer);
+                avatarMeshCombiner.renderer.sharedMesh.UploadMeshData(true);
```
[the remark this answers is not in the document]
Comment (BrianAmadori) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarRenderer.cs]: Yeah I think this can be done. Just commenting to keep you posted.

File: unity-renderer/Assets/Builder/Scripts/DCLBuilderBridge.cs [hunk 2]
```diff
@@ -579,7 +579,7 @@ private void SetCurrentScene()
             {
                 currentScene.OnEntityAdded += OnEntityIsAdded;
                 currentScene.OnEntityRemoved += OnEntityIsRemoved;
-                currentScene.metricsController = new DCLBuilderSceneMetricsController(currentScene);
+                // currentScene.metricsCounter = new DclBuilderSceneMetricsCounter(currentScene);
```
Comment (pravusjif) [inline unity-renderer/Assets/Builder/Scripts/DCLBuilderBridge.cs:582]: Let's remove the commented code

**Review by pravusjif — APPROVED** [review]
Saw the video, tested the build extensively and found no issues, great job!, let's address BiW concerns and then it's good to go.

**Review by Suduck — APPROVED** [review]
BIW is now working again! Good job!
